inversify.config.ts

```diff
@@ -0,0 +1,108 @@
+var queueUrl: string, 
+    tenantId: string,
+    s3: S3,
+    sqs: SQS;
+
+let s3Buckets = {
+    prod: 'adastra-prod-data-ingestion',
+    dev: 'adastra-dev-data-ingestion'
+}
+let bucketName = s3Buckets[stage];
+
+// NOTE: updates to the discovery service itself would require pushing a new docker image. This should still be an environment variable rather than hardcoded
+process.env['DISCOVERY_SERVICE'] = 'https://4w35qhpotd.execute-api.us-east-1.amazonaws.com/prod';
+
+var startup = () => (async () => {
+    const endpoints = await sdk.lookupService('user-management', stage);
+    process.env['USER_MANAGEMENT_URI'] = endpoints[0];
+})().then(async () => {
+    let cognitoJwt = await authManager.signIn(process.env.ASTRA_CLOUD_USERNAME, process.env.ASTRA_CLOUD_PASSWORD);
+
+    // Get IAM credentials
+    var iamCredentials = await authManager.getIamCredentials(cognitoJwt.idToken);
+
+    // Set up authenticated access to SQS
+    sqsConfig.credentials = {
+        accessKeyId: iamCredentials.AccessKeyId,
+        secretAccessKey: iamCredentials.SecretKey,
+        sessionToken: iamCredentials.SessionToken
+    };
+
+    // Set up authenticated access to S3
+    s3Config.credentials = {
+        accessKeyId: iamCredentials.AccessKeyId,
+        secretAccessKey: iamCredentials.SecretKey,
+        sessionToken: iamCredentials.SessionToken
+    };
+
+    // lookup SQS queue for this tenant
+    let credentialsBearerToken: BearerTokenCredentials = {
+        type: 'BearerToken',
+        idToken: cognitoJwt.idToken
+    };
+    let userManagementApi = new UserManagementApi(process.env.USER_MANAGEMENT_URI, REGION, credentialsBearerToken);
+    let poolListResponse = await userManagementApi.getUserPools();
+    queueUrl = poolListResponse.data[0].tenantDataIngestionQueueUrl;
+    tenantId = poolListResponse.data[0].tenant_id;
+
+}).then(() => {
+
+    container.bind<AuthManager>(TYPES.AuthManager).toConstantValue(authManager); // is this too boilerplate to be injecting everywhere? is there a way to inquire about a timed-out token?
+    container.bind<IMessageHandler>(TYPES.SendDataHandler).to(SendDataHandler);
+    container.bind<MessageFactory>(TYPES.MessageFactory).to(MessageFactory).inSingletonScope();
+    container.bind<IMessage>(TYPES.SendDataMessage).to(SendDataMessage);
+    container.bind<MessageHandlerFactory>(TYPES.MessageHandlerFactory).to(MessageHandlerFactory).inSingletonScope();
+    container.bind<S3.ClientConfiguration>(TYPES.S3Config).toConstantValue(s3Config);
+    container.bind<SQS.ClientConfiguration>(TYPES.SQSConfig).toConstantValue(s3Config);
+    container.bind<Winston.Logger>(TYPES.Logger).toConstantValue(logger);
+    container.bind<string>(TYPES.QueueUrl).toConstantValue(queueUrl);
+    container.bind<string>(TYPES.TenantId).toConstantValue(tenantId);
+    container.bind<string>(TYPES.Bucket).toConstantValue(bucketName);
+
+    container.bind<Container>(TYPES.Container).toConstantValue(container);
```

Comment:
These bindings are the meat and taters of inversify. I doubt I'm doing this all 100% correctly but it seems to be working as I would expect when running the app. Typescript decorators make this all possible.